runTests.ps1

```diff
@@ -0,0 +1,165 @@
+    [Alias('s15')]
+    [switch]$SkipVS15,
+    [Alias('sut')]
+    [switch]$SkipUnitTests,
+    [Alias('sft')]
+    [switch]$SkipFuncTests,
+    [switch]$CI
+)
+
+# For TeamCity - Incase any issue comes in this script fail the build. - Be default TeamCity returns exit code of 0 for all powershell even if it fails
+trap {
+    if ($env:TEAMCITY_VERSION) {
+        Write-Host "##teamcity[buildProblem description='$(Format-TeamCityMessage($_.ToString()))']"
+    }
+
+    Write-Host "BUILD FAILED: $_" -ForegroundColor Red
+    Write-Host "ERROR DETAILS:" -ForegroundColor Red
+    Write-Host $_.Exception -ForegroundColor Red
+    Write-Host ("`r`n" * 3)
+    exit 1
+}
+
+. "$PSScriptRoot\build\common.ps1"
+
+if (-not $Configuration) {
+    $Configuration = switch ($CI.IsPresent) {
+        $True   { 'Release' } # CI build is Release by default
+        $False  { 'Debug' } # Local builds are Debug by default
+    }
+}
+
+Write-Host ("`r`n" * 3)
+Trace-Log ('=' * 60)
+
+$startTime = [DateTime]::UtcNow
+if (-not $BuildNumber) {
+    $BuildNumber = Get-BuildNumber
+}
+Trace-Log "Test suite run #$BuildNumber started at $startTime"
+
+# Adjust version skipping if only one version installed - if VS15 is not installed, no need to specify SkipVS15
+if (-not $SkipVS14 -and -not $VS14Installed) {
+    Warning-Log "VS14 build is requested but it appears not to be installed."
+    $SkipVS14 = $True
+}
+
+if (-not $SkipVS15 -and -not $VS15Installed) {
+    Warning-Log "VS15 build is requested but it appears not to be installed."
+    $SkipVS15 = $True
+}
+
+$BuildErrors = @()
+
+Invoke-BuildStep 'Installing dotnet CLI' { Install-DotnetCLI } `
+    -ev +BuildErrors
+
+Invoke-BuildStep 'Running NuGet.Core unit-tests' {
+        param($Configuration)
+        Test-CoreProjects $Configuration
+    } `
+    -args $Configuration `
+    -skip:($SkipXProj -or $SkipUnitTests) `
+    -ev +BuildErrors
+
+Invoke-BuildStep 'Running NuGet.Core functional tests' {
+        param($Configuration)
+        Test-FuncCoreProjects $Configuration
+    } `
+    -args $Configuration `
+    -skip:($SkipXProj -or $SkipFuncTests) `
+    -ev +BuildErrors
+
+Invoke-BuildStep 'Running NuGet.Clients unit-tests - VS14 Toolset' {
+        param($Configuration)
+        Test-ClientsProjects $Configuration -ToolsetVersion 14
+    } `
+    -args $Configuration `
+    -skip:($SkipVS14 -or $SkipUnitTests) `
+    -ev +BuildErrors
+
+Invoke-BuildStep 'Running NuGet.Clients functional tests - VS14 Toolset' {
+        param($Configuration)
+        Test-FuncClientsProjects $Configuration -ToolsetVersion 14
+    } `
+    -args $Configuration `
+    -skip:($SkipVS14 -or $SkipFuncTests) `
+    -ev +BuildErrors
+
+Invoke-BuildStep 'Running NuGet.Clients tests - VS15 Toolset' {
+        param($Configuration)
+        # We don't run command line tests on VS15 as we don't build a nuget.exe for this version
+        Test-ClientsProjects $Configuration -ToolsetVersion 15 -SkipProjects 'NuGet.CommandLine.Test'
```

Comment:
I think this exclusion was pre-existing, but why *not* run these tests? It is possible we could catch extra issues here?